lib/core/src/recover/recoverer.rs

```diff
@@ -70,8 +70,13 @@ impl Recoverer {
 
         let mut preimages = HashMap::new();
         for (swap_id, claim_tx) in claim_txs_by_swap_id {
-            if let Ok(preimage) = Self::get_send_swap_preimage_from_claim_tx(swap_id, &claim_tx) {
-                preimages.insert(swap_id, preimage);
+            match Self::get_send_swap_preimage_from_claim_tx(swap_id, &claim_tx) {
```

Comment:
It seems also we are missing the validation of the preimage like we do from the status stream: https://github.com/breez/breez-sdk-liquid/blob/main/lib/core/src/send_swap.rs#L119

This alone I think is enough to make sure the transaction is a true claim.